lib/index.js

```diff
@@ -26,16 +26,18 @@ const setupHarakiri = (options) => {
   }
 
   initialized = true;
-  if (ttl) {
-    setInterval(() => {
-      for (const id in workers) {
-        if ((Date.now() - workers[id].startTime) > ttl) {
-          debug(`Terminating worker ${id} due to age`);
-          terminateWorker(id);
-        }
+  setInterval(() => {
+    debug(workers);
+    for (const id in workers) {
+      if (ttl && (Date.now() - workers[id].startTime) > ttl) {
+        debug(`Terminating worker ${id} due to age`);
+        terminateWorker(id);
+      } else if (connectionLimit && workers[id].connectionCount > connectionLimit) {
+        debug(`Terminating worker ${id} due to connection limit`);
+        terminateWorker(id);
       }
-    }, ttl / 10);
-  }
+    }
+  }, (ttl || 10000) / 10);
```

Comment:
I am not sure about this, because if I use both a ttl and a max conn, I could have a high ttl, and during busy periods well exceed my max conn.

I actually think I would utilize this by setting a ttl of like 4 hours, and a max conn of maybe 2000. Then on weekends, services would restart every 4 hours (low usage) but during the week more frequently as the max conn would be hit relatively quickly.

I think you could just cap with with a `max(..., 30000)` or something to remain in a reasonable range.

Reply:
Maybe this should just be an environment variable with a default of 30 s rather than basing it off of ttl?

Reply:
Probably, I would suggest using `min()` in that case: `Math.min(ttl, CHECK_INTERVAL)` So you can't exceed `ttl`.

Reply:
As long as `ttl` is set